api/web3/concordium.js

```diff
@@ -49,12 +55,76 @@ class Concordium {
       SchemaVersion.V2
     );
 
+    if (!account)
+    {
+      return returnValue.all_tokens;
+    }
+
     const data = returnValue.state.filter((accountInfo) => {
       return accountInfo[0].Account[0] === account;
     });
 
     return data[0][1].owned_tokens;
   }
+  async getNextNFT() {
+    return String(Object.keys(await this.listNFTs()).length + 1).padStart(
+        8,
+        "0"
+    );
+  }
+  async mintNFT(token) {
+    const accountAddress   = new AccountAddress(this.address);
+    const nextAccountNonce = await this.client.getNextAccountNonce(accountAddress);
+    const nonce            = nextAccountNonce.nonce;
+    const signer           = await buildBasicAccountSigner(this.private_key);
+
+    const contractName        = this.contractName;
+    const receiveFunctionName = "mint";
+    const receiveName         = contractName + '.' + receiveFunctionName;
+    const rawModuleSchema     = Buffer.from(fs.readFileSync('./schema.bin'));
+    const schemaVersion       = SchemaVersion.V1;
+    const parameters          = {
+      owner: {Account: [this.address]},
```

Comment:
super important: this account here needs to be received (and validated with signature) via API, so the API will define who the owner of this token will be

Reply:
Please take a look again @jatitoam